File: 프로그래머스/1/131128. 숫자 짝꿍/숫자 짝꿍.py
def solution(X, Y):
    answer = []
    
    for i in range(9, -1, -1):
        char = str(i)
        count_x = X.count(char)
        count_y = Y.count(char)

        common_count = min(count_x, count_y)

        answer.append(char * common_count)
        
    result = "".join(answer)
    
    if result == "":
        return "-1"
    
    if result[0] == "0":
        return "0"
    
    return result

File: 프로그래머스/1/131128. 숫자 짝꿍/숫자 짝꿍.js
function solution(X, Y) {
    const countX = new Array(10).fill(0);
    const countY = new Array(10).fill(0);

    for (const char of X) countX[char]++;
    for (const char of Y) countY[char]++;
    
    let result = [];

    for (let i = 9; i >= 0; i--) {
        const commonCount = Math.min(countX[i], countY[i]);
        result.push(String(i).repeat(commonCount));
    }
    
    const answer = result.join('');
    
    if (answer === "") return "-1";
    if (answer[0] === "0") return "0";
    return answer;
}